lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -657,20 +664,30 @@ final class GenericRole extends SemanticRole {
       return;
     }
 
-    // Assign one of three roles to the element: group, heading, text.
+    // Assign one of three roles to the element: group, banner, heading, text.
     //
     // - "group" is used when the node has children, irrespective of whether the
     //   node is marked as a header or not. This is because marking a group
     //   as a "heading" will prevent the AT from reaching its children.
+    // - "banner" is used to identify site-oriented content at the beginning
+    //   of each page within a website. Site-oriented content children can be
+    //   a logo, identity of the site sponsor, and site-specific-search tool.
+    //   Typically appears at the top of the page and typically spans the
+    //   full width. W3.org reference:
+    //   https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/banner.html
     // - "heading" is used when the framework explicitly marks the node as a
     //   heading and the node does not have children.
+    //   W3.org reference: https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA12
     // - If a node has a label and no children, assume is a paragraph of text.
     //   In HTML text has no ARIA role. It's just a DOM node with text inside
     //   it. Previously, role="text" was used, but it was only supported by
     //   Safari, and it was removed starting Safari 17.
     if (semanticsObject.hasChildren) {
       labelAndValue!.preferredRepresentation = LabelRepresentation.ariaLabel;
       setAriaRole('group');
+    } else if (semanticsObject.hasChildren && semanticsObject.hasFlag(ui.SemanticsFlag.isHeader)) {
+      labelAndValue!.preferredRepresentation = LabelRepresentation.domText;
+      setAriaRole('banner');
     } else if (semanticsObject.hasFlag(ui.SemanticsFlag.isHeader)) {
```

Comment:
This should probably be updated as well

Reply:
Apologies for the delay, will update that as well. Will I need to add any tests for these changes @chunhtai?